src/postAggregation.js

```diff
@@ -0,0 +1,64 @@
+'use strict'
+
+var Promise = require('q')
+var _ = require('./lodash')
+
+var aggregation = require('./aggregation')
+
+module.exports = function(service) {
+  return {
+    post: post,
+    sortByKey: sortByKey,
+    limit: limit,
+    squash: squash,
+  }
+
+  function post(query, parent, cb) {
+    query.data = parent.locked ? _.clone(parent.data) : parent.data
```

Comment:
Yeah, we should split that out.  And no, but they should be.  I'll build that in.